### PR TITLE
refactor: Update export dialog for deployments

### LINKED ISSUE #2372: Update Export dialog to add more deployment targets
Screenshot 2023-09-25 at 3 30 32 PM

Step 3 of the Export dialog is changed. A select menu is added to allow choosing from a list of deployment targets.

In the description for Step 3 there's a link to the documentation for the selected deployment target. The link text and link target should change to match the selected target.

[See the design in figma for details.](https://www.figma.com/file/xCBegXEWxROLqA1Y31z2Xo/%F0%9F%93%96-Webstudio-Design-Docs?type=design&node-id=7145-55747&mode=design)

Let me know if you have any questions.

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
## Description

fixes #2372 

Updates the export dialog with the new deployment target changes.

## Code Review

- [x] hi @kof, I need you to do
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [x] made a self-review

### CODE REVIEW

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 1]
```diff
@@ -554,24 +571,40 @@ const ExportContent = () => {
...
-              Vercel CLI
+              {humanizeString(deployTarget)}
             </StyledLink>{" "}
-            to publish your project on Vercel.
           </Text>
         </Grid>
+
+        <Select
+          fullWidth
+          css={{ zIndex: Number.MAX_SAFE_INTEGER.toString() }}
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:587]: why?

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 2]
```diff
@@ -554,24 +571,40 @@ const ExportContent = () => {
...
+          id={id}
+          value={deployTarget}
+          options={Object.keys(DeployTargets)}
+          getLabel={(value) => humanizeString(value as string)}
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:591]: why `as`

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 1 above)
Comment (JayaKrishnaNamburu) [reply apps/builder/app/builder/features/topbar/publish.tsx:587]: This select is on top of a existing popup. So, it's going behind the popup.

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 3]
```diff
@@ -554,24 +571,40 @@ const ExportContent = () => {
...
+          onChange={(value) =>
+            selectDeployTarget(value as keyof typeof DeployTargets)
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:593]: use type guard to check type example https://github.com/webstudio-is/webstudio-builder/blob/f5882256b8d542ca1f5b00e248d8e6a66b51e09f/packages/asset-uploader/src/utils/mime.ts#L40

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 2 above)
Comment (JayaKrishnaNamburu) [reply apps/builder/app/builder/features/topbar/publish.tsx:591]: Just to humanize the labels, the keys from here
Screenshot 2023-09-26 at 17 25 39

But in the UI, we should show a human version of it. LIke, if it is `cloudfare-worker` something like that in future.

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 4]
```diff
@@ -484,9 +486,24 @@ const StyledLink = styled("a", {
   },
 });
 
+type DeployTarget = "vercel" | "netlify";
+
+const DeployTargets: Record<DeployTarget, string> = {
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:491]: its not a type use "deployTargets"

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/topbar/publish.tsx:587]: seems like select should be using portal inside, we use portal 99% of the time to solve zindex issues

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 5]
```diff
@@ -484,9 +486,24 @@ const StyledLink = styled("a", {
...
+  vercel: "npx vercel",
+  netlify: "npx netlify",
+};
+
+const DeployTargetDocs: Record<DeployTarget, string> = {
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:496]: Move everything into a single place:
```
deployTargets = {
  vercel: {
   command: ''
   target: '''
  }
}
```

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/topbar/publish.tsx:587]: ah no its already using a portal, then the question is what is publish popover doing to be higher zIndex

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 6]
```diff
@@ -484,9 +486,24 @@ const StyledLink = styled("a", {
...
+  vercel: "https://vercel.com/docs/cli",
+  netlify: "https://docs.netlify.com/cli/get-started/",
+};
+
 const ExportContent = () => {
+  const id = useId();
   const npxCommand = "npx webstudio-cli";
-  const npxVercelCommand = "npx vercel";
+  const [deployTarget, selectDeployTarget] =
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:504]: setDeployTarget

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/topbar/publish.tsx:587]: seems like popover is using zIndex: theme.zIndices[1], in publish.tsx, so probably would be just using theme.zIndices[2]

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 5 above)
Comment (istarkov) [reply apps/builder/app/builder/features/topbar/publish.tsx:496]: Try to not use `Record<>` without `"noUncheckedIndexedAccess": true` in tsconfig its super error prone.
declare `as const`
```ts
const deployTarget = { .... } as const;
```

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 7]
```diff
@@ -554,24 +571,40 @@ const ExportContent = () => {
...
+              {humanizeString(deployTarget)}
             </StyledLink>{" "}
-            to publish your project on Vercel.
           </Text>
         </Grid>
+
+        <Select
+          fullWidth
+          css={{ zIndex: Number.MAX_SAFE_INTEGER.toString() }}
+          id={id}
```
Comment (istarkov) [inline apps/builder/app/builder/features/topbar/publish.tsx:588]: DO we need it at all without label

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 2 above)
Comment (istarkov) [reply apps/builder/app/builder/features/topbar/publish.tsx:591]: Try use `as const` as I wrote in other comments, look strange, should already be a string

File: apps/builder/app/builder/features/topbar/publish.tsx (hunk 7 above)
Comment (JayaKrishnaNamburu) [reply apps/builder/app/builder/features/topbar/publish.tsx:591]: I didn't get it, all the other comments are done now 👍

Comment (istarkov) [reply apps/builder/app/builder/features/topbar/publish.tsx:591]: What the reason of using id with Select only. Usually its used for labels but I dont see label

Comment (JayaKrishnaNamburu) [reply apps/builder/app/builder/features/topbar/publish.tsx]: Ah, not needed. I was trying different components from the design system. And missed removing it. Removed now

**Review by kof — APPROVED** [review]

File: apps/builder/app/builder/features/topbar/publish.tsx [hunk 8]
```diff
@@ -484,9 +486,26 @@ const StyledLink = styled("a", {
   },
 });
 
+const deployTargets = {
+  vercel: {
+    command: "npx vercel",
+    docs: "https://vercel.com/docs/cli",
+  },
+  netlify: {
+    command: "npx netlify",
```
Comment (kof) [inline apps/builder/app/builder/features/topbar/publish.tsx:495]:
```suggestion
    command: "npx netlify-cli",
```

**Review by istarkov — APPROVED** [review]

**Review by kof — APPROVED** [review]

**Pull request merged**